test/file.jl

```diff
@@ -855,6 +855,107 @@ end
 @test_throws ArgumentError download("good", "ba\0d")
 @test_throws ArgumentError download("ba\0d", "good")
 
+###################
+#     walkdir     #
+###################
+
+dirwalk = mktempdir()
+cd(dirwalk) do
+    for i=1:2
+        mkdir("sub_dir$i")
+        open("file$i", "w") do f end
+
+        mkdir(joinpath("sub_dir1", "subsub_dir$i"))
+        open(joinpath("sub_dir1", "file$i"), "w") do f end
+    end
+    open(joinpath("sub_dir2", "file_dir2"), "w")
```

Comment:
FWIW I think `touch` does this, could also be used above. As in

    touch(joinpath("sub_dir2", "file_dir2"))